src/common/NetworkManager.ts

```diff
@@ -252,12 +271,12 @@ export class NetworkManager extends EventEmitter {
         requestId
       );
 
+      this._requestIdToRequestWillBeSentEvent.set(requestId, event);
+
       if (requestPausedEvent) {
         const interceptionId = requestPausedEvent.requestId;
         this._onRequest(event, interceptionId);
         this._requestIdToRequestPausedEvent.delete(requestId);
-      } else {
```

Comment:
@jschfflr Dropping this condition is the only workaround introduced, the rest is just refactoring.